context_test.go

```diff
@@ -262,3 +263,24 @@ func TestContext_lookupFlagSet(t *testing.T) {
 		t.Fail()
 	}
 }
+
+// TestContextPropagation tests that
+// *cli.Context always has a valid
+// context.Context
+func TestContextPropagation(t *testing.T) {
+	ctx := NewContext(nil, nil, nil)
+	if ctx.Context == nil {
+		t.Fatal("expected a non nil context when no parent is present")
+	}
+	parent := NewContext(nil, nil, nil)
+	parent.Context = context.WithValue(context.Background(), "key", "val")
+	ctx = NewContext(nil, nil, parent)
```

Comment:
What about a test where parent.Context is nil, but parent not nil.

Reply:
@AudriusButkevicius i added that scenario to the bottom of this test.